### PR TITLE
Cleanup: QuestManager & TrophyManager

### PR DESCRIPTION
**Pull request by AngeloTadeucci**
Extracting helper functions to its own helper class.

### CODE REVIEW

File: MapleServer2/Tools/ConditionHelper.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        // Check if the value is in the list
+        if (conditionType.Contains(',') && IsInList(conditionType, value))
+        {
+            return true;
+        }
+
+        return long.TryParse(conditionType, out long parsedCondition) && parsedCondition == value;
+    }
+
+    public static bool IsInList(string conditionList, long value)
```
Comment (JamieDavidson) [inline MapleServer2/Tools/ConditionHelper.cs:55]: Do these need to be public? I think we could probably get away with only `IsMatching` being public?

Comment (AngeloTadeucci) [reply MapleServer2/Tools/ConditionHelper.cs:55]: just fixed

Comment (AngeloTadeucci) [reply MapleServer2/Tools/ConditionHelper.cs:55]: Wait sorry, I was thinking in making it private but decided it could be useful in the future.

Comment (JamieDavidson) [reply MapleServer2/Tools/ConditionHelper.cs:55]: I don't think those methods will ever be useful publicly. In order to figure out which one to use, you'll have to duplicate the logic of for example seeing if there's a `-` in them. I think by making those methods public, we'll be encouraging people to duplicate that logic.

Comment (AngeloTadeucci) [reply MapleServer2/Tools/ConditionHelper.cs:55]: We never know that the company has in stock for us but sure i'll make them private

**Review by JamieDavidson — APPROVED** [review]

**Pull request merged**